zeeguu/api/endpoints/sessions.py

```diff
@@ -1,11 +1,30 @@
 import flask
+from datetime import datetime
 from flask import request, make_response
 from zeeguu.core.model import Session, User
 from zeeguu.api.utils.abort_handling import make_error
 
-from zeeguu.api.utils.route_wrappers import cross_domain, with_session
+from zeeguu.api.utils.route_wrappers import cross_domain, has_session
 from . import api, db_session
 
+MAX_TIME_SESSION = 30  # Days
+
+
+def validate_session(session_object):
+    print("----------- Validating the Session based on Last use!! -----------")
+    if session_object is None:
+        flask.abort(401)
+    is_session_too_old = (
+        datetime.now() - session_object.last_use
+    ).days > MAX_TIME_SESSION
+    if is_session_too_old:
+        print("Session was too old! - logging out!")
+        db_session.delete(session_object)
+        db_session.commit()
+        flask.abort(401)
+        return False
```

Comment:
pycharm claims this code is unreachable. i guess abort is implemented as a throw?